cpp_polska/thread_local.md

```diff
@@ -0,0 +1,221 @@
+# Mało znany kuzyn statica - thread\_local
+C++11 wprowadził wiele funkcjonalności, bez których nie wyobrażamy sobie pisania nowoczesnego kodu. Należą do nich na przykład move semantics, lambda czy smart pointery. Są też ficzery, z których korzysta się raczej rzadko, ale warto je znać i wiedzieć co potrafią. 
+Jednym z takich dodatków jest słowo kluczowe **thread\_local**.
+
+# Problem
+Ostatnio natrafiłem w aplikacji wielowątkowej na problem z zapisem do bazy danych przy dużym obciążeniu. Fragment kodu odpowiedzialny za komunikację 
+z bazą danych okazał się być nieprzystosowany do pracy na wielu wątkach. Bazą danych było mongo, do połączenia używaliśmy bilbioteki mongocxx. Poniżej
+przykładowy fragment kodu: 
+
+    class MongoCoordinator {
+      public:
+        void WriteData(/* some data*/) {
+          auto collection = client_[collection_name];
+          // ... process
+        }
+
+      private: 
+        static mongocxx::instance instance_;
+        static mongocxx::client client_;
+    }
+
+    mongocxx::instance MongoCoordinator::instance_ = {}
```

Comment:
Nie wiem czy pedagogiczne jest pokazywanie przykładu z singletonem :)

Reply:
To jest wymóg biblioteki: http://mongodb.github.io/mongo-cxx-driver/mongocxx-v3/tutorial/ 
```
Before making any connections, you need to create one and only one instance of mongocxx::instance. This instance must exist for the entirety of your program
```
Ale jeśli znasz lepszy praktyczny przykład, chętnie go podmienię!

Reply:
Tak jak uzgodniliśmy: lepszym przykładem będzie jakiś logger. Dzięki!